packages/@aws-cdk/cdk/lib/app.ts

```diff
@@ -136,7 +136,30 @@ export class App extends Root {
     // add jsii runtime version
     libraries['jsii-runtime'] = getJsiiAgentVersion();
 
+    // filter out any non-CDK libraries
+    for (const name of Object.keys(libraries)) {
+      if (!filterCDKLibraries(name)) {
+        delete libraries[name];
+      }
+    }
+
     return { libraries };
+
+    function filterCDKLibraries(name: string) {
+      if (name.startsWith('@aws-cdk')) {
+        return true;
+      }
+
+      if (name === 'aws-cdk') {
+        return true;
+      }
+
+      if (name.startsWith('jsii-')) {
```

Comment:
I feel this risks including stuff that's not ours (imagine someone coding a `jsii` wrapper around some fo their internal stuff and decides to call it `jsii-our-wonderful-thing`). Maybe we should additionally inspect the package's metadata in search for the `author` field and ensure ti's AWS?

Reply:
You are right, and that's actually not needed. I've reorganized the code a bit so it basically only includes packages that begin with `@aws-cdk/`, which is much safer. `jsii-runtime` is a fake name used to represent the runtime version of jsii.